cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -222,174 +225,196 @@ func (p *BicepProvider) promptForParameter(
-
-		// location can be combined with allowedValues and with usageName metadata
-		// allowedValues == nil => all locations are allowed
-		// allowedValues != nil => only the locations in the allowedValues are allowed
-		// usageName != nil => the usageName is validated for quota for each allowed location (this is for Ai models),
-		//                     reducing the allowed locations to only those that have quota available
-		// usageName == nil => No quota validation is done
-		var allowedLocations []string
-		if param.AllowedValues != nil {
-			allowedLocations = make([]string, len(*param.AllowedValues))
-			for i, option := range *param.AllowedValues {
-				allowedLocations[i] = option.(string)
-			}
+	if isLocationParam {
+		if paramType != provisioning.ParameterTypeString {
+			return nil, fmt.Errorf("location parameter must be a string")
 		}
-		if len(azdMetadata.UsageName) > 0 {
-			if allowedLocations == nil {
-				allLocations, err := p.subscriptionManager.ListLocations(ctx, p.env.GetSubscriptionId())
-				if err != nil {
-					return nil, fmt.Errorf("listing locations: %w", err)
-				}
-				allowedLocations = make([]string, len(allLocations))
-				for i, location := range allLocations {
-					allowedLocations[i] = location.Name
+		envLoc := p.env.GetLocation()
+		if envLoc != "" {
+			// If the location is already set in the environment, use it as the default value
+			value = envLoc
+		}
+	}
+
+	if value == nil {
+		if paramType == provisioning.ParameterTypeString &&
+			(isLocationParam || (azdMetadata.Type != nil &&
+				*azdMetadata.Type == azure.AzdMetadataTypeLocation)) {
+
+			// location can be combined with allowedValues and with usageName metadata
+			// allowedValues == nil => all locations are allowed
+			// allowedValues != nil => only the locations in the allowedValues are allowed
+			// usageName != nil => the usageName is validated for quota for each allowed location (this is for Ai models),
+			//                     reducing the allowed locations to only those that have quota available
+			// usageName == nil => No quota validation is done
+			var allowedLocations []string
+			if param.AllowedValues != nil {
+				allowedLocations = make([]string, len(*param.AllowedValues))
+				for i, option := range *param.AllowedValues {
+					allowedLocations[i] = option.(string)
 				}
 			}
-			withQuotaLocations, err := p.locationsWithQuotaFor(
-				ctx, p.env.GetSubscriptionId(), allowedLocations, azdMetadata.UsageName)
-			if err != nil {
-				return nil, fmt.Errorf("getting locations with quota: %w", err)
+			if len(azdMetadata.UsageName) > 0 {
+				if allowedLocations == nil {
+					allLocations, err := p.subscriptionManager.ListLocations(ctx, p.env.GetSubscriptionId())
+					if err != nil {
+						return nil, fmt.Errorf("listing locations: %w", err)
+					}
+					allowedLocations = make([]string, len(allLocations))
+					for i, location := range allLocations {
+						allowedLocations[i] = location.Name
+					}
+				}
```

Comment:
Could we skip this altogether, since if `allowedLocations == nil`, `locationsWithQuotaFor` would effectively just use the locations from `GetResourceSkuLocations`? (assuming `p.subscriptionManager.ListLocations` is a superset of `GetResourceSkuLocations`)

Reply:
yes, that is a very good catch.

Reply:
I think we still need to add something like this in `locationsWithQuotaFor`:

```go
	if locations == nil {
		locations = azureAiServicesLocations
	}
```